src/drawtypes/layouticonset.cpp

```diff
@@ -0,0 +1,74 @@
+#include "drawtypes/layouticonset.hpp"
+
+POLYBAR_NS
+
+namespace drawtypes {
+  layouticonset::layouticonset(label_t&& default_icon) : m_default_icon(default_icon) {}
+
+  void layouticonset::add(const string& layout, const string& variant, label_t&& icon) {
+    m_layout_icons.push_back(std::make_tuple(layout, variant, icon));
+  }
+
+  label_t layouticonset::get(const string& layout, const string& variant) const {
+    // The layout, variant are matched against defined icons in that order:
+    //     1. perfect match on layout and perfect match on variant (ex: us;Colemak;<icon>)
+    //     2. perfect match on layout and case insensitive search on variant (ex: us;coLEmAk;<icon>)
+    //     3. perfect match on layout and the any variant '_' (ex: us;<icon> or us;_;<icon>)
+    //     4. no layout for icon and perfect match on variant (ex: _;Colemak;<icon>)
+    //     5. no layout for icon and case insensitive search on variant (ex: _;coLEmAk;<icon>)
+    //     6. no match at all => default icon if defined
+    bool layout_ivariant_matched = false;
+    bool layout_wildcardvariant_matched = false;
+    bool layout_novariant_matched = false;
+    bool nolayout_variant_matched = false;
+
+    auto icon = m_default_icon;
+    for (auto it : m_layout_icons) {
+      const string& icon_layout = std::get<0>(it);
+      const string& icon_variant = std::get<1>(it);
+      label_t icon_label = std::get<2>(it);
+
+      if (icon_layout == layout) {
+        if (icon_variant == variant || (icon_variant == VARIANT_NONE && variant.empty())) {
```

Comment:
We do the exact same check further down. We should just compute it once and reuse it.

Reply:
Removing VARIANT_NONE made this condition much simpler